src/core/agent/services/authService.ts

```diff
@@ -0,0 +1,106 @@
+import { AgentService } from "./agentService";
+import { AgentServicesProps, MiscRecordId } from "../agent.types";
+import { Agent } from "../agent";
+import { LoginAttempts } from "./auth.types";
+import { BasicRecord } from "../records";
+
+class AuthService extends AgentService {
+  static readonly TIME_UNIT = 60 * 1000;
+
+  constructor(agentServiceProps: AgentServicesProps) {
+    super(agentServiceProps);
+  }
+
+  static readonly LOGIN_ATTEMPT_RECORD_NOT_FOUND =
```

Comment:
Note I was going to just always create it if it doesn't exist, but actually this is OK this way :)